lib/assets/AssetsSource.js

```diff
@@ -82,4 +82,32 @@ AssetsSource.prototype.toString = function() {
   return this.srcPath + "/" + this.pattern;
 };
 
+/**
+  * Build a string suitable for caching an instance of this
+  * @returns {String} the cache key
+  */
+AssetsSource.prototype.cacheKey = function(namespace) {
+  function skipSomeKeys(key, value) {
+    // these are set to this.srcPath, which is already included in the cacheKey
+    if (key === "cwd" || key === "root") {
+      return undefined;
+    }
+    // these are added inside glob and always set to true, which happens after this string
+    // is created when glob.sync() is run in #getAssets()
+    // (see #setopts() in glob/common.js)
+    if (key === "nonegate" || key === "nocomment") {
+      return undefined;
+    }
+    return value;
+  }
+
+  return "[" +
+    "httpPrefix=" + (this.httpPrefix ? this.httpPrefix : "") +
+    ";name=" + (this.name ? this.name : (namespace ? namespace : "")) +
+    ";srcPath=" + this.srcPath +
+    ";pattern=" + this.pattern +
+    ";opts=" + JSON.stringify(this.globOpts, skipSomeKeys) +
```

Comment:
`JSON.stringify` isn't stable, so it will sometimes produce different ordering of keys in the output. I would recommend using `json-stable-stringify` instead.

Reply:
oh ok, I didn't know that - I'll change that

Reply:
ya, i found out the hard way...